app/src/sharedTest/java/org/oppia/android/app/classroom/ClassroomListFragmentTest.kt

```diff
@@ -0,0 +1,878 @@
+  }
+
+  @Test
+  fun testFragment_withAdminProfile_configChange_profileNameIsDisplayed() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
+    fakeOppiaClock.setCurrentTimeToSameDateTime(EVENING_TIMESTAMP)
+
+    composeRule.activity.recreate()
+    testCoroutineDispatchers.runCurrent()
+
+    onView(isRoot()).perform(orientationLandscape())
+
+    composeRule.onNodeWithTag(WELCOME_TEST_TAG)
+      .assertTextContains("Good evening, Admin!")
+      .assertIsDisplayed()
+  }
+
+  @Test
+  fun testFragment_morningTimestamp_goodMorningMessageIsDisplayed_withAdminProfileName() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
+    fakeOppiaClock.setCurrentTimeToSameDateTime(MORNING_TIMESTAMP)
+
+    composeRule.activity.recreate()
+    testCoroutineDispatchers.runCurrent()
+
+    composeRule.onNodeWithTag(WELCOME_TEST_TAG)
+      .assertTextContains("Good morning, Admin!")
+      .assertIsDisplayed()
+  }
+
+  @Test
+  fun testFragment_afternoonTimestamp_goodAfternoonMessageIsDisplayed_withAdminProfileName() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
+    fakeOppiaClock.setCurrentTimeToSameDateTime(AFTERNOON_TIMESTAMP)
+
+    composeRule.activity.recreate()
+    testCoroutineDispatchers.runCurrent()
+
+    composeRule.onNodeWithTag(WELCOME_TEST_TAG)
+      .assertTextContains("Good afternoon, Admin!")
+      .assertIsDisplayed()
+  }
+
+  @Test
+  fun testFragment_eveningTimestamp_goodEveningMessageIsDisplayed_withAdminProfileName() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
+    fakeOppiaClock.setCurrentTimeToSameDateTime(EVENING_TIMESTAMP)
+
+    composeRule.activity.recreate()
+    testCoroutineDispatchers.runCurrent()
+
+    composeRule.onNodeWithTag(WELCOME_TEST_TAG)
+      .assertTextContains("Good evening, Admin!")
+      .assertIsDisplayed()
+  }
+
+  @Test
+  fun testFragment_logUserInFirstTime_checkPromotedStoriesIsNotDisplayed() {
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_HEADER_TEST_TAG).assertDoesNotExist()
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_TEST_TAG).assertDoesNotExist()
+  }
+
+  @Test
+  fun testFragment_recentlyPlayedStoriesTextIsDisplayed() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_UPTIME_MILLIS)
+    storyProgressTestHelper.markInProgressSavedFractionsStory0Exp0(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    storyProgressTestHelper.markInProgressSavedRatiosStory0Exp0(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    logIntoAdminTwice()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_HEADER_TEST_TAG).onChildAt(0)
+      .assertTextContains(context.getString(R.string.recently_played_stories))
+      .assertIsDisplayed()
+  }
+
+  @Test
+  @Ignore("Temporarily ignored as the test is failing.")
```

Comment:
Probably because of the scaling problem. Try to replicate this behaviour manually on device and see if you can recreate the failure, or if it produces the desired result. You can also check the home activity if the behavior is the same.

Reply:
Fixed the test by manually replicating the steps required for the desired result.